packages/amplify-graphql-api-construct/src/internal/cdk-shim.ts

```diff
@@ -0,0 +1,145 @@
+import { Template } from '@aws-amplify/graphql-transformer-interfaces';
+import { Asset } from 'aws-cdk-lib/aws-s3-assets';
+import { CfnIncludeProps } from 'aws-cdk-lib/cloudformation-include';
+import { Construct } from 'constructs';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as os from 'os';
+
+const ROOT_STACK_FILE_NAME = 'stack.json';
+const STACKS_DIRECTORY = 'stacks';
+const SCHEMA_FILE_NAME = 'schema.graphql';
+const RESOLVERS_DIRECTORY = 'resolvers';
+
+/**
+ * This is a temporary workaround since the transformer internally synthesizes using CDK.
```

Comment:
Regarding `transformer internally synthesizes using CDK` - is this using proper CDK synthesis or derived version of Synthesizer that is basically a hack to intercept templates in memory.
Taking about this beauty https://github.com/aws-amplify/amplify-category-api/blob/c20c06d797c02f7ea2c9c9861461a091506863a7/packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts#L11 .

Reply:
No, unfortunately it's still derived. At this point the transformer has been refactored to decouple it from the Amplify core libraries (types, core utils, printer), as well as direct dependencies on files in the local amplify folder, but we haven't refactored the core to no longer apply overrides+synth. That is intended to come in a future milestone, which is why we're only targeting an alpha state with the current code-base, since there are limitations and gotchas associated w/ the current approach.